### ISSUE TITLE
Character encoding is reset when setting Content-Type

### ISSUE THREAD

**Issue by hiji** [eclipse/jetty.project#10349]
**Jetty version(s)**

12.0.0

**Jetty Environment**

ee10

**Java version/vendor** `(use: java -version)`

```
openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Temurin-17.0.8+7 (build 17.0.8+7)
OpenJDK 64-Bit Server VM Temurin-17.0.8+7 (build 17.0.8+7, mixed mode)
```

**OS type/version**

macOS 13.4.1 

**Description**

If the character encoding of HTTP response (`HttpServletResponse`) is set by `setCharacterEncoding` method and `Content-Type` that does not include charset (e.g. `text/html`) is set, the character encoding set by `setCharacterEncoding` method will be used as the `charset` of `Content-Type`.

However, if you set same `Content-Type` (e.g. `text/html`) again, the character encoding set by `setCharacterEncoding` method will be reset and a different `Content-Type` will be set.

Checking the source code, `onRemoveField` is called when setting the set header again, but if the header is `Content-Type`, the character encoding is removed. ([reference](https://github.com/eclipse/jetty.project/blob/7de4b6a0e9f5fabaab6b3ecf2505a67cdf29d1a5/jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java#L586C37-L586C37))

This is implemented in https://github.com/eclipse/jetty.project/pull/9927 in `12.0.0.beta3`, so it doesn't happen in `12.0.0.beta2` and earlier.

**How to reproduce?**

Perform the following processing in Servlet.

1. set character encoding with `response.setCharacterEncoding`.
2. set Content-Type that does not include charset with `response.setContentType`.
3. perform same process again as step 2.

For example, the following code.

```java
response.setCharacterEncoding(StandardCharsets.UTF_8.name());
response.setContentType("text/html"); // Content-Type after processing is "text/html;charset=utf-8"
response.setContentType("text/html"); // Content-Type after processing is "text/html;charset=iso-8859-1"
```

**Comment by gregw** [comment]
I've reproduced a problem, but after the second set I get "text/html;charset=null".   Anyway, looking!

**Comment by hiji** [comment]
@gregw 
Thank you for confirming.
After checking again, the charset after calling `setContentType` method was definitely `null`.
In environment I confirmed, Result was output using `getWriter` method after that, but `getCharacterEncoding` method was called in `getWriter` method, and default character encoding `iso-8859-1` was set.
([that code is here](https://github.com/eclipse/jetty.project/blob/9d908839cf388db578ecea8cbc71d75ec787ca4d/jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextResponse.java#L425))

So the correct state is as follows.

```java
response.setCharacterEncoding(StandardCharsets.UTF_8.name());
response.setContentType("text/html"); // Content-Type after processing is "text/html;charset=utf-8"
response.setContentType("text/html"); // Content-Type after processing is "text/html;charset=null"
```

**Comment by gregw** [comment]
This has been fixed by #10358
